scheduler-worker/src/main/java/gov/nasa/jpl/aerie/scheduler/worker/ListenSchedulerCapability.java

```diff
@@ -27,38 +28,42 @@ public ListenSchedulerCapability(
   }
 
-  public void registerListener() throws SQLException {
-    PGConnection connection = this.dataSource.getConnection().unwrap(PGConnection.class);
-    connection.addNotificationListener(new SchedulingRequestNotificationListener());
+  public void registerListener() {
+    new Thread(() -> {
+      try (final var connection = this.dataSource.getConnection()) {
+        try (final var listenSimulationStatusAction = new ListenSchedulingRequestStatusAction(connection)) {
+          listenSimulationStatusAction.apply();
+        } catch (final SQLException ex) {
+          throw new DatabaseException("Failed to register as LISTEN to postgres database.", ex);
+        }
 
-    try (final var listenSchedulingRequestStatusAction = new ListenSchedulingRequestStatusAction(connection)) {
-       listenSchedulingRequestStatusAction.apply();
-    } catch (final SQLException ex) {
-      throw new DatabaseException("Failed to register as LISTEN to postgres database.", ex);
-    }
-  }
-
-  private final class SchedulingRequestNotificationListener implements PGNotificationListener {
-
-    @Override
-    public void notification(final int processId, final String channelName, final String payload){
-      logger.info("Received PSQL Notification: {}, {}, {}", processId, channelName, payload);
-      final var jsonValue = Json.createReader(new StringReader(payload)).readValue();
-      final var notificationPayload = postgresSchedulingRequestNotificationP
-          .parse(jsonValue)
-          .getSuccessOrThrow();
-      try {
-        notificationQueue.put(notificationPayload);
-      } catch (InterruptedException ex) {
-        // We are unable to handle this exception here so the thread interrupt flag is set to true
-        // so that code higher up the call stack can see that an interrupt was issued.
-        Thread.currentThread().interrupt();
+        while (true) {
+          final var pgConnection = connection.unwrap(PGConnection.class);
+          final var notifications = pgConnection.getNotifications(10000);
+          if (notifications != null && notifications.length > 0) {
+            for (final var notification : notifications) {
+              final var processId = notification.getPID();
+              final var channelName = notification.getName();
+              final var payload = notification.getParameter();
+              logger.info("Received PSQL Notification: {}, {}, {}", processId, channelName, payload);
+              try (final var reader = Json.createReader(new StringReader(payload))) {
+                final var jsonValue = reader.readValue();
+                final var notificationPayload = postgresSchedulingRequestNotificationP
+                    .parse(jsonValue)
+                    .getSuccessOrThrow();
+                try {
+                  this.notificationQueue.put(notificationPayload);
+                } catch (InterruptedException e) {
+                  Thread.currentThread().interrupt();
```

Comment:
The statement `Thread.currentThread().interrupt();` doesn't do anything here, since we return from the Thread's `run` method immediately after.